From cd0cd281c43b1a71fa2d248a85c9b8ae94be713e Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 29 Nov 2022 09:29:49 +0100
Subject: [PATCH 236/954] feat(fdts): add OSPI nodes on eval board

This patch adds OSPI nodes on eval board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 16397fd5263fecee70b2b162e164cbca0586c7a5 feat(fdts): add OSPI nodes on eval board]
Change-Id: I6f480896a186e66e8766d2aae10c88fb7c617e10
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/278257
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 fdts/stm32mp257f-ev1.dts | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/fdts/stm32mp257f-ev1.dts b/fdts/stm32mp257f-ev1.dts
index bc2cc79eec..8b4648f3e9 100644
--- a/fdts/stm32mp257f-ev1.dts
+++ b/fdts/stm32mp257f-ev1.dts
@@ -46,6 +46,27 @@
 	status = "okay";
 };
 
+&ommanager {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ospi1_clk_pins_a
+		     &ospi1_io03_pins_a
+		     &ospi1_cs0_pins_a>;
+	status = "okay";
+
+	spi@40430000 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "okay";
+
+		flash0: flash@0 {
+			compatible = "jedec,spi-nor";
+			reg = <0>;
+			spi-rx-bus-width = <4>;
+			spi-max-frequency = <133000000>;
+		};
+	};
+};
+
 &sdmmc1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&sdmmc1_b4_pins_a>;
-- 
2.53.0

